app/viewmodels/checkAnswers/corrections/CorrectionReturnPeriodSummary.scala

```diff
@@ -43,9 +44,29 @@ object CorrectionReturnPeriodSummary {
           actions = Seq(
             ActionItemViewModel(
               "site.change",
-              controllers.corrections.routes.CorrectionReturnPeriodController.onPageLoad(CheckMode, answers.period, Index(0)).url)
+              controllers.corrections.routes.CorrectionReturnPeriodController.onPageLoad(CheckMode, answers.period, index).url)
               .withVisuallyHiddenText(messages("correctionReturnPeriod.change.hidden"))
           )
         )
     }
+
+  def getAllRows(answers: UserAnswers)(implicit messages: Messages): Option[SummaryListRow] = {
+    val periods = answers.get(AllCorrectionPeriodsQuery).getOrElse(List.empty).map{
+      correction => correction.correctionReturnPeriod.displayText
+    }
+    if(periods.nonEmpty){
```

Comment:
Now we have logic in this method could we add a super quick test for each side of this 'if'

Reply:
got it in the controller specs